### DISCUSSION TITLE
Access document in the "deleteOne" post hook

### DISCUSSION THREAD

**Discussion by vineetdigit** [Automattic/mongoose#10219] [Q&A] [answered]
I have defined a post hook on the MessageSchema. And I was trying to access the document that was deleted.

The hook gets called after a successful deletion, but I am not able to use the document that was deleted in the post hook.

Is following behavior expected? 

```
MessageSchema.post("deleteOne", (doc, next) => {
  console.log(doc, this);
  next();
});
```

`{ n: 1, ok: 1, deletedCount: 1 } {}`

**Comment by vkarpov15** [comment, marked as answer]
It is expected, because by default `deleteOne()` is a hook on `Query#deleteOne()`, not `Document#deleteOne()`. And when you call `Query#deleteOne()`, the document being deleted may not be in memory at the time of being deleted.

I'd recommend doing `MessageSchema.post('deleteOne', { document: true, query: false }, function() { this; /* deleted doc */ })` to register a hook for `Document#deleteOne()`, or use [`findOneAndDelete()`](https://mongoosejs.com/docs/api/model.html#model_Model.findOneAndDelete) instead.
